docker/lib/dependabot/docker/file_parser.rb

```diff
@@ -171,11 +171,15 @@ def manifest_files
       end
 
       def parse_helm(img_hash)
-        repo = img_hash.fetch("repository", nil)
         tag_value = img_hash.key?("tag") ? img_hash.fetch("tag", nil) : img_hash.fetch("version", nil)
+        return [] unless tag_value
+
+        repo = img_hash.fetch("repository", nil)
         registry = img_hash.fetch("registry", nil)
 
-        tag_details = tag_value.to_s.match(TAG_WITH_DIGEST).named_captures
+        tag_match = tag_value.to_s.match(TAG_WITH_DIGEST)
+        tag_details = tag_match&.named_captures || {}
+
```

Comment:
I think the fallback to `{}` if no named captures are found is not needed? I'm also not sure which effect it would have. If the tag content did not match `TAG_WITH_DIGEST`, we'd probably want to either raise a user error, or ignore the dependency, but I'm not sure if this change achieves that? I think I'd leave it as is for now.

Reply:
Good call! I botched the amending of my commit. I'll fix this and re-request a review 😄